src/traces/histogram/attributes.js

```diff
@@ -71,14 +71,44 @@ module.exports = {
         ].join(' ')
     },
 
-    mode: {
+    cumulative: {
+        valType: 'boolean',
+        dflt: false,
+        role: 'info',
+        description: [
+            'If true, display the cumulative distribution by summing the',
+            'binned values. Use the `direction` and `centralbin` attributes',
+            'to tune the accumulation method.'
+        ].join(' ')
+    },
+
+    direction: {
         valType: 'enumerated',
-        values: ['density', 'cumulative'],
-        dflt: 'density',
+        values: ['increasing', 'decreasing'],
+        dflt: 'increasing',
         role: 'info',
         description: [
-            ''
-        ].join('')
+            'Only applies if `cumulative=true.',
+            'If *increasing* (default) we sum all prior bins, so the result',
+            'increases from left to right. If *decreasing* we sum later bins',
+            'so the fresult decreases from left to right.'
+        ].join(' ')
+    },
+
+    currentbin: {
+        valType: 'enumerated',
+        values: ['include', 'exclude', 'half'],
+        dflt: 'include',
+        role: 'info',
+        description: [
+            'Only applies if `cumulative=true.',
+            'Sets whether the current bin is included, excluded, or has half',
+            'of its value included in the current cumulative value.',
+            '*include* is the default for compatibility with various other',
+            'tools, however it introduces a half-bin bias to the results.',
+            '*exclude* makes the opposite half-bin bias, and *half* removes',
+            'it.'
```

Comment:
Thanks for putting that option in! 

The name `currentbin` bothers me a little bit because it doesn't sound associated with `cumulative`. 

This has me thinking: maybe we should group all `cumulative` attributes into a nested object.

```js
cumulative: {
  enabled: true || false,
  direction: 'increasing' || 'decreasing',
  currentbin: 'include' || 'exclude' || 'half'
}
```

By @chriddyp's https://github.com/plotly/plotly.js/pull/1189#issuecomment-262603778 where we might extend `cumulative: true` to other trace types down the road, adding a `cumulative` nested object to scatter would be less intrusive I think.